src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFileAsSpan(statusFilePath, reusableReader, out ReadOnlySpan<char> statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                int endIndex = statusFileContents.IndexOf('\n');
+
+                // Reached end of file
+                if (endIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                ReadOnlySpan<char> value = default;
+                // endIndex - startIndex - 1 --> To slice till just before '\n'
+                // endIndex - startIndex - 4 --> To slice till just before ' kB\n'
+                switch (title.ToString())
```

Comment:
It'd be good to avoid this.  This can just be an if/else if cascade that compares SequenceEqual for each known sequence.  That's what the C# compiler would generate anyway for (IIRC) <= 5 string comparisons, and so 7 isn't that far off.  Otherwise we're going to be allocating a bunch of strings, including for fields that don't match and we don't care about.

Reply:
Done.